pkg/stores/user/cache.go

```diff
@@ -276,3 +276,9 @@ func (c *Cache) Set(ctx context.Context, u *entity.User) error {
 	c.cache.Add(u.UUID, u)
 	return nil
 }
+
+// This was written to avoid the zero value trap
+func (c *Cache) SetNotoriety(ctx context.Context, u *entity.User, notoriety int) error {
+	u.Notoriety = notoriety
+	return c.backing.SetNotoriety(ctx, u, notoriety)
```

Comment:
i know the other methods[1] have the same problem too, but (aside from not being able to transact[2]) updating the cached value before the db returned success seems incorrect. the slightly more correct version would be something like
```go
err := c.backing.SetNotoriety(ctx, u, notoriety)
if err != nil {
  return err
}
u.Notoriety = notoriety
return nil
```
[1] in particular `.Set` itself, by design.
[2] with transaction, this becomes even less clear, because the cache should be updated on successful commit, not on a successful backing.SetNotoriety.